pkg/sql/pgwire/encoding.go

```diff
@@ -153,7 +154,12 @@ func (b *readBuffer) getUint32() (uint32, error) {
 type writeBuffer struct {
 	wrapped bytes.Buffer
 	err     error
-	putbuf  [64]byte
+
+	// These two buffers are used as temporary storage. Use putbuf when the
+	// length of the required temp space is known. Use variablePutbuf when the length
+	// of the required temp space is unknown, or when a bytes.Buffer is needed.
+	putbuf         [64]byte
+	variablePutbuf bytes.Buffer
```

Comment:
I'm all ears if you have a more elegant solution to this, keeping in mind `Datum.Format`'s signature.